magpie/db.py

```diff
@@ -31,14 +32,33 @@
 configure_mappers()
 
 
-def get_db_url(username=None, password=None, db_host=None, db_port=None, db_name=None, settings=None):
-    return "postgresql://%s:%s@%s:%s/%s" % (
-        username if username is not None else get_constant("MAGPIE_POSTGRES_USER", settings, "postgres.user"),
-        password if password is not None else get_constant("MAGPIE_POSTGRES_PASSWORD", settings, "postgres.password"),
-        db_host if db_host is not None else get_constant("MAGPIE_POSTGRES_HOST", settings, "postgres.host"),
-        db_port if db_port is not None else get_constant("MAGPIE_POSTGRES_PORT", settings, "postgres.port"),
-        db_name if db_name is not None else get_constant("MAGPIE_POSTGRES_DB", settings, "postgres.db"),
-    )
+def get_db_url(username=None,   # type: Optional[Str]
+               password=None,   # type: Optional[Str]
+               db_host=None,    # type: Optional[Str]
+               db_port=None,    # type: Optional[Union[Str,int]]
+               db_name=None,    # type: Optional[Str]
+               settings=None,   # type: AnySettingsContainer
+               ):               # type: (...) -> Str
+    """Retrieve the database connection URL with provided settings."""
+    db_url = get_constant("MAGPIE_DB_URL", settings, raise_missing=False, print_missing=True, raise_not_set=False)
+    if db_url:
+        LOGGER.info("Using setting 'MAGPIE_DB_URL' for database connection.")
+    else:
+        def _get(param, kw):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Argument name "kw" doesn't conform to snake_case naming style](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650806)